### PR TITLE
Add basic SQLQueryGenerator & PostgreSQL

### PR DESCRIPTION
**Pull request by hh1680651**
- Added basic SQL query class: SQLQueryGenerator
- Added PostgreSQL's generator & connector & configuration.
- Amended SQLGenerator (extends SQLQueryGenerator).

### CODE REVIEW

File: cloudberry/neo/conf/application.conf [hunk 1]
```diff
@@ -89,10 +89,11 @@ berry.query.gap = "1 day"
 #asterixdb.url = "http://localhost:19002/aql"
 asterixdb.url = "http://localhost:19002/query/service"
 mysqldb.url = "jdbc:mysql://localhost:6603/sql?protocol=TCP&user=root"
```
Comment (JavierJia) [inline cloudberry/neo/conf/application.conf]: we should also have a `postgreSQL.url`?

File: cloudberry/zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TestQuery.scala [hunk 2]
```diff
@@ -1045,8 +1045,8 @@ object TestQuery {
        |}
     """.stripMargin)
 
-  val TwitterDataSetForSQL = TwitterDataStoreForSQL.DatasetName
-  val twitterSchemaForSQL = TwitterDataStoreForSQL.TwitterSchemaForSQL
+  val TwitterDataSetForSQL = TwitterDataStoreForMySQL.DatasetName
```
Comment (JavierJia) [inline cloudberry/zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TestQuery.scala]: `TwitterDataSetForSQL` -> `twitterDataSetForSQL`?

File: cloudberry/zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TwitterDataStoreForPostgreSQL.scala [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package edu.uci.ics.cloudberry.zion.model.impl
+
+import edu.uci.ics.cloudberry.zion.model.schema._
+
+object TwitterDataStoreForPostgreSQL {
```
Comment (JavierJia) [inline cloudberry/zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TwitterDataStoreForPostgreSQL.scala]: I think the naming cause the confusion. I would call it `TwitterDataStoreWithHashTag` and the other one is `TwitterDataStoreWithoutHashTag`. They are two different table that can be used to both db.

File: cloudberry/neo/conf/application.conf (hunk 1 above)
Comment (hh1680651) [reply cloudberry/neo/conf/application.conf]: Ok.

File: cloudberry/zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TestQuery.scala (hunk 2 above)
Comment (hh1680651) [reply cloudberry/zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TestQuery.scala]: Ok.

File: cloudberry/zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TwitterDataStoreForPostgreSQL.scala (hunk 3 above)
Comment (hh1680651) [reply cloudberry/zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/TwitterDataStoreForPostgreSQL.scala]: Ok.

**Comment by JavierJia** [comment]
#372

**Pull request merged**